tests/integration.rs

```diff
@@ -647,6 +647,33 @@ pub mod tcp {
         })
         .unwrap();
     }
+
+    #[test]
+    fn offline_server_test() {
```

Comment:
I don't see how `offlive_server_test` captures the issue. The problem happens only when the server is offline and the client gave up trying to reconnect. I don't think a simple unit test will work here. We need to move to https://github.com/testcontainers/testcontainers-rs.

I don't ask you doing this in this PR (but feel free to do it if you want 😃 )

Reply:
I think I understand what you did. You purposely choose a wrong port to trigger the issue at the first attempt. Forget what I said earlier then, it's a good idea!